### PR TITLE
Provider tally extraction script

### LINKED ISSUE #380: Initial analysis of Redis provider tallies pre & post iNaturalist ingestion
## Description

On the API, WordPress/openverse-api#1088 & WordPress/openverse-api#1113 added provider count tallying for entries that appear in search results. We would like to extract that information and analyze it to get a picture of how frequently providers show up in results pre & post turning on the new iNaturalist ingestion (WordPress/openverse-catalog#745). This will involve two steps:

1. Pulling the information from Redis
2. Analyzing the data

### Pull the info from Redis

We can use [the dead link tally script](https://github.com/WordPress/openverse/blob/108c84795289c28da4296a192a30038a0bb85f12/dead_links/dead_link_tally.py) as an example of how to connect to Redis and pull data using the `scan` function. Based on the keys, we'll want to use `match="provider*"`. We should store this information in the form of a CSV for easy analysis.

### Analyze the data

I think the easiest way to analyze this would be to write a second script (possibly using [pandas](https://pandas.pydata.org/)) which aggregates (either by average or mean or both) the provider counts per week that this has been running. Graphs that could be produced from this, like a pie chart or a percentage bar graph which shows the percent amount that each provider shows up in results. It would also be useful to gate analysis by date for this (e.g. "all data after [x] week") so we can use the same script to analyze data after iNaturalist is ingested. Graphs could be posted in this issue as well.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #380 by @AetherUnbound

## Description

This PR adds a provider tally extraction script which can be used to pull out the provider-related counts from Redis.

I also moved the `dead_links` and `load_testing` utilities into a common `utilities` directory for easier grouping within this repo.

I plan to take the CSV generated from production and throw it into Google Drive, I'll share the generated graphs on the issue once the analysis is done!

## Testing Instructions

See the README for instructions on how to run the script.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
I'm actually going to draft this for now while I work on ways of munging the data that will make it easier to construct the visualizations I'm thinking of.

**Comment by AetherUnbound** [comment]
This now generates graphs and saves the CSVs per match type and media type!
2023-02-13_provider_occurrences_image
2023-02-13_provider_appeared_in_searches_image
2023-02-13_provider_occurrences_audio
2023-02-13_provider_appeared_in_searches_audio

File: utilities/provider_tallies/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Provider Tally script
+
+This script is used to extract the per-provider counts for both total results
+and appearances within searches from the Openverse API redis.
+
+## Usage
+
+1. `pipenv install`
+2. `pipenv run python provider_tally_stats.py`
+
```
Comment (krysal) [inline utilities/provider_tallies/README.md:10]: These are the instructions for testing with the API locally, right? Doesn't it need to pass Redis credentials anywhere?

Comment (AetherUnbound) [reply utilities/provider_tallies/README.md:10]: Ah, there's a note about this on the script itself, but not in the README: https://github.com/WordPress/openverse/pull/397/files#diff-fb0b292bf6766acd757c8b72104edc2d46a78ab4e0d5738bb86e4d875965f91aR4-R6

I'll add it there too 🙂

**Comment by AetherUnbound** [comment]
Updated stats for the last few weeks:

2023-03-09_provider_occurrences_image
2023-03-09_provider_appeared_in_searches_image
2023-03-09_provider_occurrences_audio
2023-03-09_provider_appeared_in_searches_audio

**Review by sarayourfriend — APPROVED** [review]
Awesome! So cool to see this data collected for several weeks and now visualised.

**Comment by sarayourfriend** [comment]
Also holy smokes, flickr is _so_ dominant :cold_sweat:

**Comment by AetherUnbound** [comment]
It really is 😳

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**